core/agents/github_journey_agent.py

```
@@ -174,4 +217,506 @@ def manage_labels(self,
+            await self.setup_branch_protection(repo["name"], "main")
+            
+            # Initialize with basic files
+            await self.initialize_repository(repo["name"], config)
+            
+            return repo
+        except Exception as e:
+            logger.error(f"Repository setup failed: {str(e)}")
+            raise HTTPException(
+                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                detail=f"Repository setup failed: {str(e)}"
+            )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)